docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +326,220 @@ recursion was explicitly requested.
 * If `{:tailrecursion true}` was specified but the code does not allow it,
 an error message is given.
 
-### 22.1.15. timeLimitMultiplier
+### 22.1.13. `{:timeLimitMultiplier X}`
 This attribute may be placed on a method or function declaration
 and has an integer argument. If `{:timeLimitMultiplier X}` was
 specified a `{:timelimit Y}` attributed is passed on to Boogie
 where `Y` is `X` times either the default verification time limit
 for a function or method, or times the value specified by the
 Boogie `timelimit` command-line option.
 
-### 22.1.16. trigger
+### 22.1.14. `{:trigger}` {#sec-trigger}
 Trigger attributes are used on quantifiers and comprehensions.
 They are translated into Boogie triggers.
 
-### 22.1.17. typeQuantifier
+### 22.1.15. `{:typeQuantifier}`
 The `{:typeQuantifier}` attribute must be used on a quantifier if it
 quantifies over types.
 
 
-## 22.2. Boogie Attributes
-Use the Boogie "/attrHelp" option to get the list of attributes
-that Boogie recognizes and their meaning. Here is the output at
-the time of this writing. Dafny passes attributes that have
-been specified to Boogie.
+## 22.2. Boogie-specific verification attributes
+Dafny passes verification attributes that have been specified to its dependency Boogie.
+Use the `/attrHelp` option to get the list of attributes
+recognized by Boogie and their meaning.
+
+### 22.2.1. Verification attributes on top-level declarations
+
+#### 22.2.1.1. `{:ignore}`
+Ignore the declaration (after checking for duplicate names).
+
+#### 22.2.1.2. `{:extern}` {#sec-extern}
+
+If a ``ClassDecl``, a ``MethodDecl`` or a ``FunctionDecl`` has an `{:extern}` attribute,
+then:
+* The compiler is going to emit nothing for it, it assumes it exists in the linked files
```

Comment:
I don't think this is true in all cases - at least some of the compilers will allow you to declare a body, and the effect of `{:extern}` is only to avoid mangling the name so that other linked files can reference the compiled symbol reliably.

You might want to leave documenting `{:extern}` better out of scope for this already-large PR as it's a bit of a rathole.

Reply:
I modified "The compiler _generally_ emits nothing for it"
I still prefer to document `{:extern}` now rather than postponing it to another PR. We can always refine it later. The sooner some correct information is available and some wrong information is removed, the better.